nerfstudio/process_data/process_data_utils.py

```diff
@@ -469,3 +479,72 @@ def save_mask(
         cv2.imwrite(str(mask_path_i), mask_i)
     CONSOLE.log(":tada: Generated and saved masks.")
     return mask_path / "mask.png"
+
+
+def generate_planar_projections_from_equirectangular(
+    image_dir: Path,
+    planar_image_size: Tuple[int, int],
+    samples_per_im: int,
+) -> Path:
+    """Generate planar projections from an equirectangular image.
+
+    Args:
+        image_dir: The directory containing the equirectangular image.
+        planar_image_size: The size of the planar projections [width, height].
+        samples_per_im: The number of samples to take per image.
+    returns:
+        The path to the planar projections directory.
+    """
+
+    device = torch.device("cuda")
+
+    fov = 120
+    yaw_pitch_pairs = []
+    if samples_per_im == 8:
+        fov = 120
+        for i in np.arange(-180, 180, 90):
+            yaw_pitch_pairs.append((i, 0))
+        for i in np.arange(-180, 180, 180):
+            yaw_pitch_pairs.append((i, 45))
+        for i in np.arange(-180, 180, 180):
+            yaw_pitch_pairs.append((i, -45))
+    elif samples_per_im == 14:
+        fov = 110
+        for i in np.arange(-180, 180, 60):
+            yaw_pitch_pairs.append((i, 0))
+        for i in np.arange(-180, 180, 90):
+            yaw_pitch_pairs.append((i, 45))
+        for i in np.arange(-180, 180, 90):
+            yaw_pitch_pairs.append((i, -45))
+
+    equi2pers = Equi2Pers(height=planar_image_size[1], width=planar_image_size[0], fov_x=fov, mode="bilinear")
+    frame_dir = image_dir
+    output_dir = image_dir / "planar_projections"
+    output_dir.mkdir(exist_ok=True)
+    num_ims = len(os.listdir(frame_dir))
+    progress = Progress(
+        TextColumn("[bold blue]Generating Planar Images", justify="right"),
+        BarColumn(),
+        TaskProgressColumn(show_speed=True),
+        ItersPerSecColumn(suffix="equirect frames/s"),
+        TimeRemainingColumn(elapsed_when_finished=True, compact=True),
+    )
+
+    curr_im = 0
+    with progress:
+        for i in progress.track(os.listdir(frame_dir), description="", total=num_ims):
+            if i.endswith(".jpg") or i.endswith(".png") or i.endswith(".jpeg") or i.endswith(".JPG"):
```

Comment:
`i.lower().endswith(('.jpg', '.jpeg', '.png'))`

Reply:
Also add ".JPG" since those images aren't recognized as ".jpg"

Reply:
I think that's why matt did i.lower() so we wouldn't have to also do .JPG